cpp/src/distance/detail/pairwise_matrix/dispatch_l2_expanded_half_float_float_int.cu

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION.
```

Comment:
All new files should have 2024 as the year. Please make this change in all new files introduced in this PR.